// @inkeep/ai-api/models/components/assistantmessage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssistantMessage$ = exports.AssistantMessage$outboundSchema = exports.AssistantMessage$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const recordscited_js_1 = require("./recordscited.js");
/** @internal */
exports.AssistantMessage$inboundSchema = z.object({
    role: z.literal("assistant").optional(),
    content: z.string(),
    records_cited: recordscited_js_1.RecordsCited$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "records_cited": "recordsCited",
    });
});
/** @internal */
exports.AssistantMessage$outboundSchema = z.object({
    role: z.literal("assistant").default("assistant"),
    content: z.string(),
    recordsCited: recordscited_js_1.RecordsCited$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        recordsCited: "records_cited",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AssistantMessage$;
(function (AssistantMessage$) {
    /** @deprecated use `AssistantMessage$inboundSchema` instead. */
    AssistantMessage$.inboundSchema = exports.AssistantMessage$inboundSchema;
    /** @deprecated use `AssistantMessage$outboundSchema` instead. */
    AssistantMessage$.outboundSchema = exports.AssistantMessage$outboundSchema;
})(AssistantMessage$ || (exports.AssistantMessage$ = AssistantMessage$ = {}));
