// @inkeep/ai-api/models/components/chatresult.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatResult$ = exports.ChatResult$outboundSchema = exports.ChatResult$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const assistantmessage_js_1 = require("./assistantmessage.js");
/** @internal */
exports.ChatResult$inboundSchema = z.object({
    chat_session_id: z.string(),
    message: assistantmessage_js_1.AssistantMessage$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "chat_session_id": "chatSessionId",
    });
});
/** @internal */
exports.ChatResult$outboundSchema = z.object({
    chatSessionId: z.string(),
    message: assistantmessage_js_1.AssistantMessage$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        chatSessionId: "chat_session_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ChatResult$;
(function (ChatResult$) {
    /** @deprecated use `ChatResult$inboundSchema` instead. */
    ChatResult$.inboundSchema = exports.ChatResult$inboundSchema;
    /** @deprecated use `ChatResult$outboundSchema` instead. */
    ChatResult$.outboundSchema = exports.ChatResult$outboundSchema;
})(ChatResult$ || (exports.ChatResult$ = ChatResult$ = {}));
