// @inkeep/ai-api/models/components/chatresultmessagechunkevent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatResultMessageChunkEvent$ = exports.ChatResultMessageChunkEvent$outboundSchema = exports.ChatResultMessageChunkEvent$inboundSchema = void 0;
const z = __importStar(require("zod"));
const messagechunk_js_1 = require("./messagechunk.js");
/** @internal */
exports.ChatResultMessageChunkEvent$inboundSchema = z.object({
    event: z.literal("message_chunk").optional(),
    data: z.string().transform((v, ctx) => {
        try {
            return JSON.parse(v);
        }
        catch (err) {
            ctx.addIssue({
                code: z.ZodIssueCode.custom,
                message: `malformed json: ${err}`,
            });
            return z.NEVER;
        }
    }).pipe(messagechunk_js_1.MessageChunk$inboundSchema),
});
/** @internal */
exports.ChatResultMessageChunkEvent$outboundSchema = z.object({
    event: z.literal("message_chunk").default("message_chunk"),
    data: messagechunk_js_1.MessageChunk$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ChatResultMessageChunkEvent$;
(function (ChatResultMessageChunkEvent$) {
    /** @deprecated use `ChatResultMessageChunkEvent$inboundSchema` instead. */
    ChatResultMessageChunkEvent$.inboundSchema = exports.ChatResultMessageChunkEvent$inboundSchema;
    /** @deprecated use `ChatResultMessageChunkEvent$outboundSchema` instead. */
    ChatResultMessageChunkEvent$.outboundSchema = exports.ChatResultMessageChunkEvent$outboundSchema;
})(ChatResultMessageChunkEvent$ || (exports.ChatResultMessageChunkEvent$ = ChatResultMessageChunkEvent$ = {}));
