// @inkeep/ai-api/models/components/chatsessioninput.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatSessionInput$ = exports.ChatSessionInput$outboundSchema = exports.ChatSessionInput$inboundSchema = void 0;
const z = __importStar(require("zod"));
const message_js_1 = require("./message.js");
/** @internal */
exports.ChatSessionInput$inboundSchema = z.object({
    guidance: z.nullable(z.string()).optional(),
    context: z.nullable(z.string()).optional(),
    messages: z.array(message_js_1.Message$inboundSchema),
    tags: z.array(z.string()).optional(),
});
/** @internal */
exports.ChatSessionInput$outboundSchema = z.object({
    guidance: z.nullable(z.string()).optional(),
    context: z.nullable(z.string()).optional(),
    messages: z.array(message_js_1.Message$outboundSchema),
    tags: z.array(z.string()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ChatSessionInput$;
(function (ChatSessionInput$) {
    /** @deprecated use `ChatSessionInput$inboundSchema` instead. */
    ChatSessionInput$.inboundSchema = exports.ChatSessionInput$inboundSchema;
    /** @deprecated use `ChatSessionInput$outboundSchema` instead. */
    ChatSessionInput$.outboundSchema = exports.ChatSessionInput$outboundSchema;
})(ChatSessionInput$ || (exports.ChatSessionInput$ = ChatSessionInput$ = {}));
