// @inkeep/ai-api/models/components/citation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Citation$ = exports.Citation$outboundSchema = exports.Citation$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const record_js_1 = require("./record.js");
/** @internal */
exports.Citation$inboundSchema = z.object({
    number: z.number().int().optional(),
    record: record_js_1.RecordT$inboundSchema,
    hit_url: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "hit_url": "hitUrl",
    });
});
/** @internal */
exports.Citation$outboundSchema = z.object({
    number: z.number().int().optional(),
    record: record_js_1.RecordT$outboundSchema,
    hitUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        hitUrl: "hit_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Citation$;
(function (Citation$) {
    /** @deprecated use `Citation$inboundSchema` instead. */
    Citation$.inboundSchema = exports.Citation$inboundSchema;
    /** @deprecated use `Citation$outboundSchema` instead. */
    Citation$.outboundSchema = exports.Citation$outboundSchema;
})(Citation$ || (exports.Citation$ = Citation$ = {}));
