// @inkeep/ai-api/models/components/messagechunk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageChunk$ = exports.MessageChunk$outboundSchema = exports.MessageChunk$inboundSchema = exports.Four$ = exports.Four$outboundSchema = exports.Four$inboundSchema = exports.Four = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
var Four;
(function (Four) {
    Four["Stop"] = "stop";
    Four["Length"] = "length";
    Four["ContentFilter"] = "content_filter";
})(Four || (exports.Four = Four = {}));
/** @internal */
exports.Four$inboundSchema = z.nativeEnum(Four);
/** @internal */
exports.Four$outboundSchema = exports.Four$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Four$;
(function (Four$) {
    /** @deprecated use `Four$inboundSchema` instead. */
    Four$.inboundSchema = exports.Four$inboundSchema;
    /** @deprecated use `Four$outboundSchema` instead. */
    Four$.outboundSchema = exports.Four$outboundSchema;
})(Four$ || (exports.Four$ = Four$ = {}));
/** @internal */
exports.MessageChunk$inboundSchema = z.object({
    chat_session_id: z.nullable(z.string()).optional(),
    content_chunk: z.string(),
    finish_reason: z.nullable(z.any()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "chat_session_id": "chatSessionId",
        "content_chunk": "contentChunk",
        "finish_reason": "finishReason",
    });
});
/** @internal */
exports.MessageChunk$outboundSchema = z.object({
    chatSessionId: z.nullable(z.string()).optional(),
    contentChunk: z.string(),
    finishReason: z.nullable(z.any()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        chatSessionId: "chat_session_id",
        contentChunk: "content_chunk",
        finishReason: "finish_reason",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MessageChunk$;
(function (MessageChunk$) {
    /** @deprecated use `MessageChunk$inboundSchema` instead. */
    MessageChunk$.inboundSchema = exports.MessageChunk$inboundSchema;
    /** @deprecated use `MessageChunk$outboundSchema` instead. */
    MessageChunk$.outboundSchema = exports.MessageChunk$outboundSchema;
})(MessageChunk$ || (exports.MessageChunk$ = MessageChunk$ = {}));
