// @inkeep/ai-api/models/components/record.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordT$ = exports.RecordT$outboundSchema = exports.RecordT$inboundSchema = void 0;
const z = __importStar(require("zod"));
const recordtype_js_1 = require("./recordtype.js");
/** @internal */
exports.RecordT$inboundSchema = z.object({
    type: recordtype_js_1.RecordType$inboundSchema,
    url: z.nullable(z.string()).optional(),
    title: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    breadcrumbs: z.nullable(z.array(z.string())).optional(),
});
/** @internal */
exports.RecordT$outboundSchema = z.object({
    type: recordtype_js_1.RecordType$outboundSchema,
    url: z.nullable(z.string()).optional(),
    title: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    breadcrumbs: z.nullable(z.array(z.string())).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RecordT$;
(function (RecordT$) {
    /** @deprecated use `RecordT$inboundSchema` instead. */
    RecordT$.inboundSchema = exports.RecordT$inboundSchema;
    /** @deprecated use `RecordT$outboundSchema` instead. */
    RecordT$.outboundSchema = exports.RecordT$outboundSchema;
})(RecordT$ || (exports.RecordT$ = RecordT$ = {}));
