// @inkeep/ai-api/models/components/recordtype.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordType$ = exports.RecordType$outboundSchema = exports.RecordType$inboundSchema = exports.RecordTypeEnumerated$ = exports.RecordTypeEnumerated$outboundSchema = exports.RecordTypeEnumerated$inboundSchema = exports.RecordTypeEnumerated = void 0;
const z = __importStar(require("zod"));
var RecordTypeEnumerated;
(function (RecordTypeEnumerated) {
    RecordTypeEnumerated["Documentation"] = "documentation";
    RecordTypeEnumerated["Site"] = "site";
    RecordTypeEnumerated["DiscoursePost"] = "discourse_post";
    RecordTypeEnumerated["GithubIssue"] = "github_issue";
    RecordTypeEnumerated["GithubDiscussion"] = "github_discussion";
    RecordTypeEnumerated["StackoverflowQuestion"] = "stackoverflow_question";
    RecordTypeEnumerated["DiscordForumPost"] = "discord_forum_post";
    RecordTypeEnumerated["DiscordMessage"] = "discord_message";
    RecordTypeEnumerated["CustomQuestionAnswer"] = "custom_question_answer";
})(RecordTypeEnumerated || (exports.RecordTypeEnumerated = RecordTypeEnumerated = {}));
/** @internal */
exports.RecordTypeEnumerated$inboundSchema = z.nativeEnum(RecordTypeEnumerated);
/** @internal */
exports.RecordTypeEnumerated$outboundSchema = exports.RecordTypeEnumerated$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RecordTypeEnumerated$;
(function (RecordTypeEnumerated$) {
    /** @deprecated use `RecordTypeEnumerated$inboundSchema` instead. */
    RecordTypeEnumerated$.inboundSchema = exports.RecordTypeEnumerated$inboundSchema;
    /** @deprecated use `RecordTypeEnumerated$outboundSchema` instead. */
    RecordTypeEnumerated$.outboundSchema = exports.RecordTypeEnumerated$outboundSchema;
})(RecordTypeEnumerated$ || (exports.RecordTypeEnumerated$ = RecordTypeEnumerated$ = {}));
/** @internal */
exports.RecordType$inboundSchema = z.union([exports.RecordTypeEnumerated$inboundSchema, z.string()]);
/** @internal */
exports.RecordType$outboundSchema = z.union([exports.RecordTypeEnumerated$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RecordType$;
(function (RecordType$) {
    /** @deprecated use `RecordType$inboundSchema` instead. */
    RecordType$.inboundSchema = exports.RecordType$inboundSchema;
    /** @deprecated use `RecordType$outboundSchema` instead. */
    RecordType$.outboundSchema = exports.RecordType$outboundSchema;
})(RecordType$ || (exports.RecordType$ = RecordType$ = {}));
